GoogleForCreators/web-stories-wp
# Update pageToken to have opaque semantics in common/reducer.js

## Summary

Update pageToken to have opaque semantics in common/reducer.js.

This removes the assumption that pageToken (formally pagingNum) is always an integer that represents a page number. This allows local and media3p pagination to behave in a consistent manner, for example:

local: `pageToken == <page number>`
coverr: `pageToken == <page number>`
unsplash: `pageToken == <next page url>`
tenor: `pageToken == <page token>`

## To-do

- Write tests for local/reducer.js and common/reducer.js (https://github.com/google/web-stories-wp/issues/2512).
- QA review, or add a feature flag, but this should not be necessary

## User-facing changes

No user-facing changes.

## Testing Instructions

- Open the media pane to verify that it appears.
- Scroll down to verify a page of results in returned.
- Scroll to the last page to make sure it loads correctly.
- Upload an image successfully.
- Upload a video successfully.
- Insert an image into the canvas.
